File: src/engine/commands/eatcommand.cpp
#include "eatcommand.h"

#include "util.h"


#define super Command

EatCommand::EatCommand(QObject *parent) :
    super(parent) {

    setDescription("Eat an item in the current room or your inventory.\n"
                   "\n"
                   "Example: eat cake");
}

EatCommand::~EatCommand() {
}

void EatCommand::execute(Player *player, const QString &command) {

    super::prepareExecute(player, command);

    GameObjectPtr item = takeObject(player->inventory() + currentRoom()->items());
    if (!requireSome(item, "Eat what?")) {
        return;
    }

    if (!item->hasTrigger("oneat")) {
        send("You cannot eat that.");
        return;
    }

    if (!item->invokeTrigger("oneat", player)) {
        return;
    }

    item->setDeleted();
}

File: data/commands/eatcommand.js
function EatCommand() {
    Command.call(this);

    this.setDescription("Eat an item in the current room or your inventory.\n" +
                        "\n" +
                        "Example: eat cake");
}

EatCommand.prototype = new Command();
EatCommand.prototype.constructor = EatCommand;

EatCommand.prototype.execute = function(player, command) {

    this.prepareExecute(player, command);

    var pool = player.inventory.concat(player.currentRoom.items);
    var item = this.takeObject(pool);
    if (!this.requireSome(item, "Eat what?")) {
        return;
    }

    if (!item.hasTrigger("oneat")) {
        player.send("You cannot eat that.");
        return;
    }

    if (!item.invokeTrigger("oneat", player)) {
        return;
    }

    item.setDeleted();
};

CommandRegistry.registerCommand("eat", new EatCommand());
